### PR TITLE
Reworking WorkPosts

### PR DESCRIPTION
**Pull request by davidwessman**
Adding admin namespace
Rewriting all views.
Editing abilities and adding controller tests.

### CODE REVIEW

File: app/controllers/admin/work_posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    work_post = WorkPost.find(params[:id])
+
+    work_post.destroy!
+    redirect_to admin_work_posts_path, notice: alert_destroy(WorkPost)
+  end
+
+  private
+
+  def authorize
+    authorize!(:manage, WorkPost)
```
Comment (jforberg) [inline app/controllers/admin/work_posts_controller.rb]: Manage is not optimal for checking. Prefer using one of "modify", "update" or adding a custom one.

File: app/models/work_post.rb [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+  scope :published, -> { visible.by_deadline.by_published }
+
+  scope :target, -> (target) { where(target_group: target) }
+  scope :field, -> (field) { where(field: field) }
+  scope :kind, -> (kind) { where(kind: kind) }
+
+  def self.companies
+    where.not(company: nil).where.not(company: '').
+      order(:company).
+      pluck(:company).uniq
```
Comment (jforberg) [inline app/models/work_post.rb]: Here you could let the database do the work and write select(:company).uniq.pluck(:company) instead.

File: app/models/work_post.rb [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
+  end
+
+  def self.kinds
+    where.not(kind: nil).where.not(kind: '').
+      order(:kind).
+      pluck(:kind).uniq
+  end
+
+  def to_s
+    title
```
Comment (jforberg) [inline app/models/work_post.rb]: alias :to_s :title

**Comment by jforberg** [comment]
Good work but some comments. -1

**Comment by davidwessman** [comment]
@jforberg Thank you! Great feedback.

Fixed the comments now.

File: app/models/work_post.rb (hunk 3 above)
Comment (davidwessman) [inline app/models/work_post.rb]: It seems this only works if `title` is a method.

Comment (jforberg) [inline app/models/work_post.rb]: Ok, so rails is (ab)using #missing_method then. In that case your solution is better.

**Pull request merged**